### PR TITLE
feat(suite-native): improve mobile accessibility

### LINKED ISSUE #10502: App accesibility
Accessibility report from google play warns about couple of buttons. 
image

### PR DESCRIPTION
**Pull request by juriczech**
## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/10502

### CODE REVIEW

File: suite-native/atoms/src/Radio.tsx [hunk 1]
```diff
@@ -64,8 +64,6 @@ export const Radio = ({
         <TouchableOpacity
             disabled={isDisabled}
             onPress={() => onPress(value)}
-            accessibilityRole="radio"
```
Comment (PeKne) [inline suite-native/atoms/src/Radio.tsx:67]: Why did you remove these two lines? I don't know accessibility API very well, so this is not request for change, I'm just curious.

File: suite-native/module-dev-utils/src/screens/DemoScreen.tsx [hunk 2]
```diff
@@ -267,28 +266,17 @@ export const DemoScreen = () => {
...
                                 key="first"
                                 value="first"
+                                accessibilityLabel="First"
                                 onPress={handleRadioPress}
                                 isChecked={radioChecked === 'first'}
                             />
                             <Radio
                                 key="second"
                                 value="second"
+                                accessibilityLabel="First"
```
Comment (PeKne) [inline suite-native/module-dev-utils/src/screens/DemoScreen.tsx]:
```suggestion
                                accessibilityLabel="Second"
```

But i know this is demo screen, so its not big deal at all.

**Review by PeKne — APPROVED** [review]

File: suite-native/atoms/src/Radio.tsx (hunk 1 above)
Comment (juriczech) [reply suite-native/atoms/src/Radio.tsx:67]: Radio is always used with a wrapper that's triggering the value switch. So in select for example, there is `SelectItem` which triggers the radio change. I added the `accessibilityRole` to the wrapper component because it's needed there and since it's there, I believe it is redundant here. Furthermore, you would only get the role, but no label, which I added to the wrapper as well, so I think it's better this way. Radio is currently only used in `SelectItem` and `SelectableListItem` (which was never used so I deleted it).

I'm a little bit unsure about this but believe this is better. If you agree I would stick with this and see if google reports it or not.

File: suite-native/module-dev-utils/src/screens/DemoScreen.tsx (hunk 2 above)
Comment (juriczech) [reply suite-native/module-dev-utils/src/screens/DemoScreen.tsx]: Yeah, I just copy pasted this and then thought it's just demo screen and doesn't matter too much so didn't want to make new commit just for this :D 

But I can commit your suggestion, thanks
